js/ui/button_group.js

```diff
@@ -0,0 +1,252 @@
+             * @extends Action
+             * @type function(e)
+             * @type_function_param1 e:object
+             * @type_function_param1_field4 addedItems:array<any>
+             * @type_function_param1_field5 removedItems:array<any>
+             * @action
+             */
+            onSelectionChanged: null
+        });
+    },
+
+    _initTemplates() {
+        this.callBase();
+
+        this._defaultTemplates["item"] = new BindableTemplate((($container, data) => {
+            const buttonOptions = !isPlainObject(data) ? { text: String(data) } : data;
+            this._createComponent($container, Button, extend({}, buttonOptions, this._getBasicButtonOptions()));
+        }), ["text", "type", "icon"], this.option("integrationOptions.watchMethod"));
+    },
+
+    _initMarkup() {
+        this.setAria("role", "buttongroup");
+        this.$element().addClass(BUTTON_GROUP_CLASS);
+        this._renderButtons();
+        this._syncSelectionOptions();
+        this.callBase();
+    },
+
+    _fireSelectionChangeEvent: function(addedItems, removedItems) {
+        this._createActionByOption("onSelectionChanged", {
+            excludeValidators: ["disabled", "readOnly"]
+        })({ addedItems: addedItems, removedItems: removedItems });
+    },
+
+    _getBasicButtonOptions() {
+        return {
+            type: this.option("buttonType"),
+            focusStateEnabled: false,
+            hoverStateEnabled: this.option("hoverStateEnabled"),
+            activeStateEnabled: this.option("activeStateEnabled")
+        };
+    },
+
+    _renderButtons() {
+        const $buttons = $("<div>")
+            .addClass(BUTTON_GROUP_WRAPPER_CLASS)
+            .appendTo(this.$element());
+
+        this._buttonsCollection = this._createComponent($buttons, ButtonCollection, {
+            selectionMode: this.option("selectionMode"),
+            items: this.option("items"),
+            keyExpr: this.option("keyExpr"),
+            itemTemplate: this._getTemplateByOption("itemTemplate"),
+            scrollingEnabled: false,
+            selectedItemKeys: this.option("selectedItemKeys"),
+            selectedIndexes: this.option("selectedIndexes"),
+            focusStateEnabled: this.option("focusStateEnabled"),
+            accessKey: this.option("accessKey"),
+            tabIndex: this.option("tabIndex"),
+            noDataText: "",
+            selectionRequired: this.option("selectionMode") === "single",
+            onSelectionChanged: (e) => {
+                this._syncSelectionOptions();
+                this._fireSelectionChangeEvent(e.addedItems, e.removedItems);
+            }
+        });
+    },
+
+    _syncSelectionOptions() {
+        this._setOptionSilent("selectedIndexes", this._buttonsCollection._getSelectedItemIndices());
+        this._setOptionSilent("selectedItemKeys", this._buttonsCollection.option("selectedItemKeys"));
+    },
+
+    _setOptionSilent: function(name, value) {
```

Comment:
I think it's time to move this to the Widget from the CollectionWidget :)

Reply:
This best make as a separate task.